releasenotes/notes/check_hostname_is_usable-dc4b99e6f7242b34.yaml

```diff
@@ -0,0 +1,17 @@
+# Each section from every releasenote are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    Prevent a containerized agent from using the OS hostname as datadog agent hostname.
+    Datadog agent hostname should be stable and should reflect the identity of the host they are monitoring.
+    Container ID or kubernetes POD names are not suitable for that purpose.
+    That’s why the agent should avoid using the OS hostname when it is running in a dedicated UTS namespace.
+    Note that the agent is relying on many other smart ways of discovering the real hostname of the host
+    (EC2/AWS metadata server, docker daemon, kubernetes node names)
+    This change only acts as a safety belt when all those smart ways of discovering the hostname are not usable.
```

Comment:
Let's try to make this upgrade note as actionable as possible for users. I.e something like:

>    Starting with this version, the containerized Agent never chooses the OS hostname as its hostname when it is running in a dedicated UTS namespace. This is done in order to avoid picking container IDs or kubernetes POD names as hostnames, since these identifiers do not reflect the identity of the host they run on.
>
>    This change only affects you if [...] [here, add some additional info about how users can know if they are affected by the change, if the above info is not enough. And more importantly, write something about what users should do if they're affected by the change].

What do you think?

(if needed the PR can be merged as-is, and this release note can be changed later)